wrangler: cannot find module [plugin hybrid-nodejs_compat]

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

Whenever I run `wrangler dev`(miniflare) with `nodejs_compat` flag, it shows the following outputs indicating it couldn't find some modules, then halts.

```sh
 ⛅️ wrangler 4.38.0 (update available 4.43.0)
─────────────────────────────────────────────
Your Worker has access to the following bindings:
Binding          Resource      Mode
env.TEST         Browser       local

╭──────────────────────────────────────────────────────────────────────╮
│  [b] open a browser [d] open devtools [c] clear console [x] to exit  │
╰──────────────────────────────────────────────────────────────────────╯

✘ [ERROR] Build failed with 1 error:

  ✘ [ERROR] Cannot find module
  '/nix/store/6gzg36sba31lgan92znnjhv756rjmfam-wrangler-4.38.0/lib/packages/wrangler/node_modules/@cloudflare/unenv-preset/dist/index.mjs'
  imported from
  /nix/store/6gzg36sba31lgan92znnjhv756rjmfam-wrangler-4.38.0/lib/packages/wrangler/wrangler-dist/cli.js
  [plugin hybrid-nodejs_compat]
```

### Steps to reproduce

1. clone the repository https://github.com/abiriadev/wrangler-node-compat.
2. cd into it. if you use `direnv`, run `direnv allow`. if not, run `nix develop`.
3. run `wrangler dev`.

### Expected behaviour

When I tried to run the same command on aother os, or even inside docker container, it runs well.

### Screenshots

_No response_

### Relevant log output

```console
--- 2025-10-17T02:22:40.474Z debug
🪵  Writing logs to "~/.config/.wrangler/logs/wrangler-2025-10-17_02-22-40_348.log"
---

--- 2025-10-17T02:22:40.474Z debug
Metrics dispatcher: Source Key not provided. Be sure to initialize before sending events {"deviceId":"-","event":"wrangler command started","timestamp":1760667760474,"properties":{"amplitude_session_id":0,"amplitude_event_id":0,"wranglerVersion":"4.38.0","osPlatform":"Linux","osVersion":"#1-NixOS SMP PREEMPT_DYNAMIC Mon Oct  6 09:17:53 UTC 2025","nodeVersion":22,"isFirstUsage":false,"configFileType":"jsonc","isCI":false,"isPagesCI":false,"isWorkersCI":false,"isInteractive":true,"hasAssets":false,"argsUsed":[],"argsCombination":"","command":"wrangler dev","args":{"xRemoteBindings":true,"latest":true,"noBundle":false,"remote":false,"r":false,"testScheduled":false,"xVectorizeBindToProd":false,"xImagesLocalMode":false}}}
---

--- 2025-10-17T02:22:40.476Z debug
.env file not found at "/tmp/wrangler-node-compat/.env". Continuing... For more details, refer to https://developers.cloudflare.com/workers/wrangler/system-environment-variables/
---

--- 2025-10-17T02:22:40.477Z debug
.env file not found at "/tmp/wrangler-node-compat/.env.local". Continuing... For more details, refer to https://developers.cloudflare.com/workers/wrangler/system-environment-variables/
---

--- 2025-10-17T02:22:40.483Z log

 ⛅️ wrangler 4.38.0 (update available 4.43.0)
─────────────────────────────────────────────
---

--- 2025-10-17T02:22:40.509Z debug
local dev variables file not found at ".dev.vars". Continuing... For more details, refer to https://developers.cloudflare.com/workers/wrangler/system-environment-variables/
---

--- 2025-10-17T02:22:40.509Z debug
.env file not found at "/tmp/wrangler-node-compat/.env". Continuing... For more details, refer to https://developers.cloudflare.com/workers/wrangler/system-environment-variables/
---

--- 2025-10-17T02:22:40.509Z debug
.env file not found at "/tmp/wrangler-node-compat/.env.local". Continuing... For more details, refer to https://developers.cloudflare.com/workers/wrangler/system-environment-variables/
---

--- 2025-10-17T02:22:40.517Z log
Your Worker has access to the following bindings:
---
```

### Additional context

Note: the problem here is `nodejs_compat` flag. Without this flag, everything works fine. However, I need this option in some cases, e.g., for [puppeteer bindings](https://developers.cloudflare.com/browser-rendering/platform/wrangler/).

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.51, NixOS, 25.05 (Warbler), 25.05.20251010.7e297dd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.5`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/7wf5s8skd51jfiym9j6mahxdk6b4ka9b-source`

### Notify maintainers

@seanrmurphy @dezren39 @ryand56 @ezrizhu

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/wr/wrangler/package.nix
+++ b/pkgs/by-name/wr/wrangler/package.nix
@@ -17,13 +17,13 @@
 }:
 stdenv.mkDerivation (finalAttrs: {
   pname = "wrangler";
-  version = "4.38.0";
+  version = "4.48.0";
 
   src = fetchFromGitHub {
     owner = "cloudflare";
     repo = "workers-sdk";
     rev = "wrangler@${finalAttrs.version}";
-    hash = "sha256-4avSdp68ecDnY5ZhtVCjq9+u8bmNRpq2XTbnapmY+S0=";
+    hash = "sha256-0ren1A943mMzsAxFiUb2GUR9GFhFJ1Oen0Q4badAVKM=";
   };
 
   pnpmDeps = pnpm_9.fetchDeps {
@@ -34,7 +34,7 @@ stdenv.mkDerivation (finalAttrs: {
       postPatch
       ;
     fetcherVersion = 2;
-    hash = "sha256-oavyEIDsGCsX0cZu0fI3vpjFc/BwHNre9hNantu/wCk=";
+    hash = "sha256-n5nKP6BLw0xMgIecFVNt/vjfdQmO3HiI0vrqAtpA4ms=";
   };
   # pnpm packageManager version in workers-sdk root package.json may not match nixpkgs
   postPatch = ''
@@ -71,6 +71,9 @@ stdenv.mkDerivation (finalAttrs: {
   # so I simply removed it
   postBuild = ''
     mv packages/vitest-pool-workers packages/~vitest-pool-workers
+
+    NODE_ENV="production" pnpm --filter unenv-preset run build
+    NODE_ENV="production" pnpm --filter workers-utils run build
     NODE_ENV="production" pnpm --filter workers-shared run build
     NODE_ENV="production" pnpm --filter miniflare run build
     NODE_ENV="production" pnpm --filter wrangler run build
@@ -87,7 +90,7 @@ stdenv.mkDerivation (finalAttrs: {
     runHook preInstall
     mkdir -p $out/{bin,lib}
     mv packages/~vitest-pool-workers packages/vitest-pool-workers
-    cp -r {fixtures,packages,node_modules,vendor} $out/lib
+    cp -r {fixtures,packages,node_modules} $out/lib
     cp -r tools $out/lib/tools
     rm -rf node_modules/typescript node_modules/eslint node_modules/prettier node_modules/bin node_modules/.bin node_modules/**/bin node_modules/**/.bin
     rm -rf $out/lib/**/bin $out/lib/**/.bin

